rpm: drop support for unsupported distros

smithfarm opened the pull request:
Replaces #6225 which was targeting infernalis.

Fixes: http://tracker.ceph.com/issues/13445
References: http://tracker.ceph.com/issues/15486

smithfarm commented:
Passes gitbuilder.

ktdreyer commented:
Really great!
```
Reviewed-by: Ken Dreyer <kdreyer@redhat.com>
```
@branto1 care to review?

branto1 commented on ceph.spec.in:
```diff
@@ -37,16 +37,16 @@
+restorecon -R %{_bindir}/ceph-mds > /dev/null 2>&1; \
+restorecon -R %{_bindir}/radosgw > /dev/null 2>&1; \
 restorecon -R /etc/rc\.d/init\.d/ceph > /dev/null 2>&1; \
 restorecon -R /etc/rc\.d/init\.d/radosgw > /dev/null 2>&1; \
```
Here, we can drop the rc.d/init.d lines as well as they contain the deprecated init scripts.

branto1 commented on ceph.spec.in:
```diff
@@ -830,8 +830,8 @@ if [ $1 = 0 ] ; then
-      /usr/bin/systemctl --no-reload disable $SERVICE > /dev/null 2>&1 || :
-      /usr/bin/systemctl stop $SERVICE > /dev/null 2>&1 || :
+      %{_bindir}/systemctl --no-reload disable $SERVICE > /dev/null 2>&1 || :
+      %{_bindir}/systemctl stop $SERVICE > /dev/null 2>&1 || :
```
Here (and in several other chunks), I'm not completely sure about this. People usually use the hard-coded paths in post scripts although I'm not sure what is the reason for that.

branto1 commented:
Mostly lgtm although I'm not sure about the last two patches -- I commented on the next to last one and I know nothing about the resource agents so that is why I'm not sure about the last one.

smithfarm commented:
@branto1 The purpose of the OCF commit is to finish the job explained in http://tracker.ceph.com/issues/14828

smithfarm commented:
Changelog:
* rebased
* added `References:` line to "Drop ceph Resource Agent" commit
* added a commit dropping the sysvinit lines from `relabel_files()` (as per @branto1)

smithfarm commented:
@branto1 Can you find out what the reason is for writing out the path in the scriptlets, instead of using the macro?

Not using a macro - in even one place - would seem to defeat the purpose of using the macro in the first place.

branto1 commented:
@smithfarm: I think the idea for not using the macros in rpm scripts is that we do not know where the other binaries were installed -- it is not guaranteed that they were built with the same set of macros as we are being built. This can be especially true for things that are not standard in the distro -- you might want to create your own build of ceph and put it in e.g. /usr/local/. If you overrode the macros, your rpm scripts would constantly fail.

You can see e.g. in examples in [1] that they don't use these macros for binaries that are outside of the scope of the package and they do not use them there on purpose.

[1] https://fedoraproject.org/wiki/Packaging:Scriptlets

smithfarm commented:
@branto1 Good point! I added a commit - please take a look.

smithfarm commented:
@branto1 I'm not sure about https://github.com/ceph/ceph/pull/8720/commits/93bb2fc77c866f83958322c213cad012a451d03d#diff-ca40628478ff1274b73dcd40c384e7f0R1140 though

branto1 commented:
@smithfarm: That line should be fine. AFAIK, the general rule of thumb is to use hard-coded paths for things outside the scope of the package and macros for all the files that are handled by the package (i.e. appear in the file lists). The missing full path for ln binary should also be quite ok, especially for things like 'ln' that are standard on all distros -- these scriptlets get evaluated by bash, I'd not be too concerned that there are any distros out there that don't have 'ln' (or 'mkdir') in PATH.

smithfarm commented:
Changelog:
* rebase
* s/ceph.spec.in:/rpm:/ in all commit messages
* new commit "rpm: do a full make check when --with-tests is given"
* reworked the "replace literal paths with RPM macros" commit to really apply the rule described by @branto1
* revised "drop dead conditionals" commit to drop "%if 0%{?el6}" part of junit conditional
* re-pushed to gitbuilder

smithfarm commented:
Changelog:
* added Debian bits to "Drop ceph Resource Agent" commit

smithfarm commented:
I'm going to split the "Drop ceph Resource Agent" commit off into a separate commit.

smithfarm commented:
Changelog:
* moved the Resource Agent commit to #8818 

@liewegas This is another try at #6225 . . . since Jewel dropped support for sysvinit, would it be OK to flag this for backport to Jewel?

ktdreyer commented:
I'm :+1: for backporting this to Jewel. It will help to make future cherry-picks easier if the packaging is aligned (particularly this early in Jewel's life).

liewegas commented:
Sounds ok to me!  The only sysvinit we still have to worry about is debian; nothing I know of on the rpm side of the house.

smithfarm commented:
Passes gitbuilder. I will do some basic testing so I can assert that it doesn't break anything obvious.

branto1 commented:
I've just re-checked the whole patch-set (full diff). Great work, +1 for merging this and back-porting it to jewel.

liewegas commented:
This conflicts with the removal of the rm -f lines from the test ec plugins.  I thinkt eh conflict resolution is pretty trivial, but I think it'll be easier to backport if it's rebased first, then merged.

smithfarm commented:
Changelog
* remove rm -f lines from the test ec plugins
* found 6 or so instances of $RPM_BUILD_ROOT that I had missed; changed them to %{buildroot}

smithfarm commented:
Will merge after gitbuilder confirmation.
